### PR TITLE
Honour listall param for list templates api

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR makes changes to honour listall param for list templates api and also adds the listall parameter to deployVM in the UI.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
clgtm

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java [hunk 1]
```diff
@@ -4337,7 +4337,7 @@ private Pair<List<TemplateJoinVO>, Integer> searchForTemplatesInternal(ListTempl
         Account caller = CallContext.current().getCallingAccount();
         Long parentTemplateId = cmd.getParentTemplateId();
 
-        boolean listAll = false;
+        boolean listAll = cmd.listAll();
```
Comment (rohityadavcloud) [inline server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:4340]: should we check for who is making the request (for example, admin vs other roles)

**Comment by slavkap** [comment]
I'm not sure how to properly test this, but here are the results from a few scenarios:
1 - I've created a project and downloaded a template. When I open the deploy VM wizard I can see it. 
2 - I've created another project, and when I open the wizard I see the template of the first project
3 - when I open the default view and try to deploy a VM I see only the template from the first project, but I have a few more when I list them from the Image tab
4 - I've deleted the projects and have only the default view, and there aren't any templates listed when using the Deploy VM wizard

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java (hunk 1 above)
Comment (vishesh92) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:4340]: Let me verify this. This is similar to what is happening for list ISOs. And is being used only to decide `ListProjectResourcesCriteria` in both the APIs.

**Comment by vishesh92** [comment]
> I'm not sure how to properly test this, but here are the results from a few scenarios: 1 - I've created a project and downloaded a template. When I open the deploy VM wizard I can see it. 2 - I've created another project, and when I open the wizard I see the template of the first project 3 - when I open the default view and try to deploy a VM I see only the template from the first project, but I have a few more when I list them from the Image tab 4 - I've deleted the projects and have only the default view, and there aren't any templates listed when using the Deploy VM wizard

Thanks @slavkap . This is probably a UI issue. Let me check & fix this.

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java (hunk 1 above)
Comment (sureshanaparti) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:4340]: @vishesh92 checked this?

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Review by slavkap — APPROVED** [review]
Code LGTM, I tested the scenarios that I mentioned above.
I'm not sure what should be the behavior for the templates in the community section in the Deploy VM wizard. If I have a project and download a template, then create another project, the second project has access to the template of the first one via the community tab

**Comment by sureshanaparti** [comment]
Verified listing of templates in default view and projects.

**Pull request merged**